functional/test-env

```diff
@@ -0,0 +1,23 @@
+# The -X format changed from go1.4 -> go1.5
+function go_linker_dashX {
+  local version=$(go version)
+  local regex="go([0-9]+).([0-9]+)."
+  if [[ $version =~ $regex ]]; then
+    if [ ${BASH_REMATCH[1]} -eq "1" -a ${BASH_REMATCH[2]} -le "4" ]; then
+      echo "$1 \"$2\""
+    else
+      echo "$1=$2"
+    fi
+  else
+    echo "could not determine Go version"
+    exit 1
+  fi
+}
+
+export GOBIN=${PWD}/bin
+export GOPATH=${PWD}/gopath
```

Comment:
I would say if some of these environment are shared between the build and test, then just make a single copy in the root directory of fleet, then source it from there. Not sure if currently it's the case but it would be better if we start the functional tests from the root directory too then cd into functional ? at least one would be able to start them from root directory.

Thank you!